src/services/acspDataService.ts

```diff
@@ -29,4 +27,28 @@ export class AcspDataService {
             return Promise.reject(error);
         }
     }
+
+    typeOfBusinessChange (session:Session, acspData: AcspData, selectedOption: string): AcspData {
+        if (acspData.typeOfBusiness !== selectedOption) {
+            session.deleteExtraData(USER_DATA);
+            const clearedAcspData: AcspData = {
+                ...acspData,
+                typeOfBusiness: selectedOption,
+                registeredOfficeAddress: undefined,
+                roleType: undefined,
+                verified: undefined,
+                businessName: undefined,
+                workSector: undefined,
+                amlSupervisoryBodies: undefined,
+                companyDetails: undefined,
+                companyAuthCodeProvided: undefined,
+                howAreYouRegisteredWithAml: undefined,
+                applicantDetails: undefined
+            };
+            acspData = clearedAcspData;
+        } else {
+            acspData.typeOfBusiness = selectedOption;
```

Comment:
Also if selected option is not changed then we can just return acspData as we received it.

Reply:
Done